src/tests/HDFFile.cxx

```diff
@@ -66,6 +66,116 @@ class T_CommandHandler : public testing::Test {
     return event_index[i2] == cue_index;
   }
 
+  static void create_static_file_with_hdf_output_prefix() {
+    using namespace FileWriter;
+    using std::array;
+    using std::vector;
+    using std::string;
+    MainOpt &main_opt = *g_main_opt.load();
+    std::string hdf_output_prefix = "tmp-relative-output";
+    {
+#ifdef _MSC_VER
+      // NOTE just a suggestion, not tested on windows so far, we have no
+      // windows here.
+      _mkdir(hdf_output_prefix.c_str());
+#else
+      mkdir(hdf_output_prefix.c_str(), 0777);
+#endif
+    }
+    {
+      std::string jsontxt =
+          fmt::format(R""({{"hdf-output-prefix": "{}"}})"", hdf_output_prefix);
+      rapidjson::Document cfg;
+      cfg.Parse(jsontxt.c_str());
+      main_opt.config_file = merge(cfg, main_opt.config_file);
+      main_opt.hdf_output_prefix = hdf_output_prefix;
+    }
+    rapidjson::Document json_command;
+    {
+      using namespace rapidjson;
+      auto &j = json_command;
+      auto &a = j.GetAllocator();
+      j.SetObject();
+      Value nexus_structure;
+      nexus_structure.SetObject();
+
+      Value children;
+      children.SetArray();
+
+      {
+        Value g1;
+        g1.SetObject();
+        g1.AddMember("type", "group", a);
+        g1.AddMember("name", "some_group", a);
+        Value attr;
+        attr.SetObject();
+        attr.AddMember("NX_class", "NXinstrument", a);
+        g1.AddMember("attributes", attr, a);
+        Value ch;
+        ch.SetArray();
+        {
+          {
+            Document jd;
+            jd.Parse(
+                R""({
+                  "type": "dataset",
+                  "name": "value",
+                  "values": 42.24,
+                  "attributes": {"units":"degree"}
+                })"");
+            ch.PushBack(Value().CopyFrom(jd, a), a);
+          }
+        }
+        g1.AddMember("children", ch, a);
+        children.PushBack(g1, a);
+      }
+      nexus_structure.AddMember("children", children, a);
+      j.AddMember("nexus_structure", nexus_structure, a);
+      {
+        Value v;
+        v.SetObject();
+        v.AddMember("file_name", StringRef("tmp-file-with-hdf-prefix.h5"), a);
+        j.AddMember("file_attributes", v, a);
+      }
+      j.AddMember("cmd", StringRef("FileWriter_new"), a);
+      j.AddMember("job_id", StringRef("000000000dataset"), a);
+    }
+
+    auto cmd = json_to_string(json_command);
+    auto fname = get_string(&json_command, "file_attributes.file_name");
+    ASSERT_GT(fname.v.size(), 8);
+
+    FileWriter::CommandHandler ch(main_opt, nullptr);
+    Msg msg;
+    msg.data = (char *)cmd.data();
+    msg.size = cmd.size();
+    ch.handle(msg);
+    ASSERT_EQ(ch.file_writer_tasks.size(), (size_t)1);
+    {
+      string cmd("{\"recv_type\":\"FileWriter\", "
```

Comment:
`cmd` hides your previous `cmd` in upper scope. I think this code block should be extracted to a separate function, especially as it is repeated elsewhere.